contrib_bots/lib/thesaurus.py

```diff
@@ -0,0 +1,86 @@
+# See zulip/contrib_bots/lib/readme.md for instructions on running this code.
+from __future__ import print_function
+import logging
+from PyDictionary import PyDictionary as Dictionary
+
+#Uses Python's Dictionary module
+#  pip install PyDictionary
+def get_thesaurus_result(original_content):
+
+    search_keyword = original_content.strip().split(' ', 1)[1]
+    if original_content.startswith('@synonym') and search_keyword != 'help':
+        try:
+            synonym = Dictionary.synonym(search_keyword)
+        except Exception as e:
+            logging.exception(e)
+            return e
+        return ', '.join(synonym)
+    elif original_content.startswith('@antonym') and search_keyword != 'help':
+        try:
+            antonym = Dictionary.antonym(search_keyword)
+        except Exception as e:
+            logging.exception(e)
+            return e
+        return ', '.join(antonym)
+    elif search_keyword == 'help':
+        help_message = "To use this bot, start messages with either \
+                        @synonym (to get the synonyms of a given word) \
+                        or @antonym (to get the antonyms of a given word). \
+                        Phrases are not accepted so only use single words \
+                        to search. For example you could search '@synonym hello' \
+                        or '@antonym goodbye'."
+        return help_message
+
+class ThesaurusHandler(object):
+    '''
+    This plugin allows users to enter a word in zulip
+    and get synonyms, and antonyms, for that word sent
+    back to the context (stream or private) in which
+    it was sent. It looks for messages starting with
+    @synonym or @antonym.
+    '''
+
+    def usage(self):
+        return '''
+            This plugin will allow users to get both synonyms
+            and antonyms for a given word from zulip. Use
+            '@thesaurus help' for more usage
+            information. Users should preface messages with @synonym
+            or @antonym.
+            '''
+
+    def triage_message(self, message, client):
+        # return True if we want to (possibly) response to this message
+
+        original_content = message['content']
+
+        # This next line of code is defensive, as we
+        # never want to get into an infinite loop of posting synonyms
+        # and antonyms for the same search term.
+        if message['display_recipient'] == 'thesaurus':
```

Comment:
This next line of code really only prevents an error if somebody uses the thesaurus stream, so I'd just remove it.